src/app/modules/account/pages/registration/components/user-registration/user-registration.component.ts

```diff
@@ -28,9 +28,44 @@ import {Account} from "../../../../../../models/account.model";
   templateUrl: "./user-registration.component.html",
   styleUrls: ["./user-registration.component.scss"],
   standalone: true,
-  imports: [IonicModule, CommonModule, FormsModule],
+  imports: [IonicModule, CommonModule, FormsModule,ReactiveFormsModule],
 })
-export class UserRegistrationComponent {
+export class UserRegistrationComponent implements OnInit{
   @Input() account?: Partial<Account>;
-  constructor() {}
+  registrationForm: FormGroup;
+
+  constructor(private fb: FormBuilder) {
+    this.registrationForm = this.fb.group({
+      name: ['', Validators.required],
+      tagline:[''],
+      Description :[''],
+      phoneNumber:['', Validators.pattern(/^-?\d+$/)],
+      alternateEmail:['',[Validators.email]],
+      mailingAddress:[''],
+      emergencyContactNumber:['', Validators.pattern(/^-?\d+$/)],
+      preferredMethodOfContact :[''],
+      occupation:[''],
+      employerName:[''],
+      workExperience:[''],
+      skillsAndExpertise:[''],
+      currentJobTitle:[''],
+      linkedInProfile:[''],
+      educationalBackground:['']
+    });
+  }
+
+  ngOnInit(){
```

Comment:
You can pre-populate the form with something like this.
```
  loadFormData() {
    if (!this.account) return;
    this.editAccountForm.patchValue({
      name: this.account.name,
      email: this.account.email,
      description: this.account.description,
      tagline: this.account.tagline,
      language: this.account.language,
      userDetails: {
        dateOfBirth: this.account.userDetails?.dateOfBirth
          .toDate()
          .toISOString(),
      },
      address: {
        ...this.account.address,
      },
      phone: {
        ...this.account.phone,
      },
    });
  }
  ```
  There's a good chance you won't have this data, but you should at least have the email address to pre-populate.

Reply:
Please checkout the `edit-user-profile.ts` page for the latest version of the update code.

Reply:
I'm going to resolve this comment as I think we'll need to copy over the `group-registration` component for this component and base this component on that, as it's now fully built and has all the needed functionality for this component.